src/tests/test_subclass.py

```diff
@@ -104,8 +104,10 @@ def test_interface():
     assert ob.bar("bar", 2) == "bar/bar"
     assert FunctionsTest.test_bar(ob, "bar", 2) == "bar/bar"
 
-    x = FunctionsTest.pass_through(ob)
-    assert id(x) == id(ob)
+    # pass_through will convert from InterfaceTestClass -> IInterfaceTest,
+    # causing a new wrapper object to be created. Hence id will differ.
+    x = FunctionsTest.pass_through_interface(ob)
+    assert id(x) != id(ob)
```

Comment:
I don't know whether this could cause any surprises.

We still have that `x == y` and that `len(set([x,y]) == 1`, so the objects can still be used as dict keys etc with the same behavior as before. But perhaps there are other effects I have not thought of.